catalog/resource_provider.go

```diff
@@ -51,15 +57,19 @@ func (a ProvidersAPI) updateProvider(ci *ProviderInfo) error {
 }
 
 func ResourceProvider() *schema.Resource {
-	providerSchema := common.StructToSchema(ProviderInfo{}, func(m map[string]*schema.Schema) map[string]*schema.Schema {
+	providerInfoSchema := common.StructToSchema(ProviderInfo{}, func(m map[string]*schema.Schema) map[string]*schema.Schema {
+		m["authentication_type"].ValidateFunc = validation.StringInSlice([]string{"TOKEN"}, false)
```

Comment:
and where `recipient_profile_str` will be coming from?...

Reply:
you need to write `m["recipient_profile_str"].DiffSuppressFunc`, which will suppress the diff when platform returns zero value and we have some in the config. this is "inverted" behavior from `tf:"suppress_diff"` and has to be hand-coded